api/net/Areas/Subscriber/Models/NotificationModel.cs

```diff
@@ -0,0 +1,131 @@
+    /// </summary>
+    public NotificationType NotificationType { get; set; }
+
+    /// <summary>
+    /// get/set - The foreign key to the template.
+    /// </summary>
+    public int TemplateId { get; set; }
+
+    /// <summary>
+    /// get/set - The Razor templates to generate the notification.
+    /// </summary>
+    public NotificationTemplateModel? Template { get; set; }
+
+    /// <summary>
+    /// get/set - When to resend the notification.
+    /// </summary>
+    public ResendOption Resend { get; set; } = ResendOption.Never;
+
+    /// <summary>
+    /// get/set - Foreign key to user who owns this notification.
+    /// </summary>
+    public int? OwnerId { get; set; }
+
+    /// <summary>
+    /// get/set - Whether this notification is public to all users.
+    /// </summary>
+    public bool IsPublic { get; set; }
+
+    /// <summary>
+    /// get/set - Whether this notification is automatically run when the content is indexed.
+    /// </summary>
+    public bool AlertOnIndex { get; set; }
+
+    /// <summary>
+    /// get/set - The default filter for this notification.
+    /// </summary>
+    public FilterSettingsModel Settings { get; set; } = new();
+
+    /// <summary>
+    /// get/set - The settings for this notification.
+    /// </summary>
+    public JsonDocument Query { get; set; } = JsonDocument.Parse("{}");
+
+    /// <summary>
+    /// get - List of users who are subscribed to this notification (many-to-many).
+    /// </summary>
+    public virtual IEnumerable<UserModel> Subscribers { get; set; } = Array.Empty<UserModel>();
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of an NotificationModel.
+    /// </summary>
+    public NotificationModel() { }
+
+    /// <summary>
+    /// Creates a new instance of an NotificationModel, initializes with specified parameter.
+    /// </summary>
+    /// <param name="entity"></param>
+    /// <param name="options"></param>
+    public NotificationModel(Entities.Notification entity, JsonSerializerOptions options) : base(entity)
+    {
+        this.NotificationType = entity.NotificationType;
+        this.TemplateId = entity.TemplateId;
+        this.Template = entity.Template != null ? new NotificationTemplateModel(entity.Template, options) : null;
+        this.Resend = entity.Resend;
+        this.OwnerId = entity.OwnerId;
+        this.IsPublic = entity.IsPublic;
+        this.AlertOnIndex = entity.AlertOnIndex;
+        this.Settings = JsonSerializer.Deserialize<FilterSettingsModel>(entity.Settings, options) ?? new();
+        this.Query = entity.Query;
+    }
+    #endregion
+
+    #region Methods
+    /// <summary>
+    /// Creates a new instance of a Notification object.
+    /// </summary>
+    /// <param name="options"></param>
+    /// <param name="includeTemplate"></param>
+    /// <returns></returns>
+    public Entities.Notification ToEntity(JsonSerializerOptions options, bool includeTemplate = false)
```

Comment:
Are you ever going to send this model back to the API?  I doubt it.  Why include this method?